### ISSUE TITLE
[Bug Report] TinyScaler installation fails

### ISSUE THREAD

**Issue by nicky-lenaers** [Farama-Foundation/TinyScaler#23]
**Describe the bug**
Installing `tinyscaler` fails.

**Code example**
`pyproject.toml`:
```toml
[tool.poetry]
name = "example"
version = "0.0.0"
description = ""
authors = []
readme = "README.md"

[tool.poetry.dependencies]
python = "~3.10"
torch = "2.0.0"
stable-baselines3 = "2.0.0"
fastapi = "^0.100.0"
gunicorn = "^21.2.0"
httpx = "^0.24.1"
uvicorn = {version = "^0.23.1", extras = ["standard"]}
gymnasium = "0.28.1"
pettingzoo = {version = "1.23.1", extras = ["classic"]}
supersuit = "3.9.0"
azureml-core = "^1.52.0"

[[tool.poetry.source]]
name = "PyPI"
priority = "primary"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"

```
Run `poetry install`.

**System Info**
Describe the characteristic of your environment:
- MacBook Air M2 2022
- macOS Ventura 13.5
- Poetry 1.5.1
- Python 3.10.12

**Additional context**
Error in console:
```sh
  • Installing tinyscaler (1.2.6): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel
  
  running bdist_wheel
  running build
  running build_ext
  building 'tinyscaler' extension
  creating build
  creating build/temp.linux-aarch64-cpython-310
  creating build/temp.linux-aarch64-cpython-310/src
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Isrc -I/tmp/tmpgtxmcz0v/.venv/include -I/usr/local/python/3.10.12/include/python3.10 -c src/tinyscaler.c -o build/temp.linux-aarch64-cpython-310/src/tinyscaler.o -mfpu=neon
  gcc: error: unrecognized command-line option ‘-mfpu=neon’
  error: command '/usr/bin/gcc' failed with exit code 1
  

  at /usr/local/py-utils/venvs/poetry/lib/python3.10/site-packages/poetry/installation/chef.py:147 in _prepare
      143│ 
      144│                 error = ChefBuildError("\n\n".join(message_parts))
      145│ 
      146│             if error is not None:
    → 147│                 raise error from None
      148│ 
      149│             return path
      150│ 
      151│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with tinyscaler (1.2.6) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "tinyscaler (==1.2.6)"'.
```
Then, running the verification command `pip wheel --use-pep517 "tinyscaler (==1.2.6)"` yields:
```sh
pip wheel --use-pep517 "tinyscaler (==1.2.6)"
Collecting tinyscaler==1.2.6
  Downloading tinyscaler-1.2.6.tar.gz (127 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 127.7/127.7 kB 3.2 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy>=1.21.0 (from tinyscaler==1.2.6)
  Obtaining dependency information for numpy>=1.21.0 from https://files.pythonhosted.org/packages/b1/39/3f88e2bfac1fb510c112dc0c78a1e7cad8f3a2d75e714d1484a044c56682/numpy-1.25.2-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl.metadata
  Downloading numpy-1.25.2-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl.metadata (5.6 kB)
Downloading numpy-1.25.2-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (14.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.2/14.2 MB 9.4 MB/s eta 0:00:00
Saved ./numpy-1.25.2-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
Building wheels for collected packages: tinyscaler
  Building wheel for tinyscaler (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for tinyscaler (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [10 lines of output]
      running bdist_wheel
      running build
      running build_ext
      building 'tinyscaler' extension
      creating build
      creating build/temp.linux-aarch64-cpython-310
      creating build/temp.linux-aarch64-cpython-310/src
      gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Isrc -I/workspaces/example/.venv/include -I/usr/local/python/3.10.12/include/python3.10 -c src/tinyscaler.c -o build/temp.linux-aarch64-cpython-310/src/tinyscaler.o -mfpu=neon
      gcc: error: unrecognized command-line option ‘-mfpu=neon’
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tinyscaler
Failed to build tinyscaler
ERROR: Failed to build one or more wheels

```

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/TinyScaler/issues) in the repo (**required**)

**Comment by 222464** [comment]
Hey, sorry I only see this now. TinyScaler assumes neon instructions are available on ARM platforms. It seems GCC isn't finding the option for the M2. Can you try removing lines [18-19](https://github.com/Farama-Foundation/TinyScaler/blob/f6704f97f6583a08e4c796f5be4f60b86e61ff6d/setup.py#L18) from setup.py and installing from the repository with `pip install .` to see if it enabled by default? If that doesn't work, in the worst case I can make it use the "No manual SIMD" fallback code for M2.

**Comment by kapilrampalli** [comment]
I just ran into this error as well while installing trying to install TinyScaler in a Docker image on the M1. After a bit of digging, I found out that GCC doesn't support the `mfpu` flag in the AArch64 backend. AArch64 CPUs are required to support NEON instructions, so the flag would end up being redundant. I think this could be fixed by excluding the SIMD flag when the architecture is `aarch64` or `arm64`. 

Docs:
https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html
https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html
https://stackoverflow.com/questions/29851128/gcc-arm64-aarch64-unrecognized-command-line-option-mfpu-neon
